EKF/ekf_helper.cpp

```diff
@@ -496,17 +496,7 @@ bool Ekf::realignYawGPS()
 			// adjust the quaternion covariances estimated yaw error
 			increaseQuatYawErrVariance(sq(asinf(sineYawError)));
 
-			// reset the corresponding rows and columns in the covariance matrix and set the variances on the magnetic field states to the measurement variance
-			clearMagCov();
-
-			if (_control_status.flags.mag_3D) {
-				for (uint8_t index = 16; index <= 21; index ++) {
-					P[index][index] = sq(_params.mag_noise);
-				}
-
-				// save covariance data for re-use when auto-switching between heading and 3-axis fusion
```

Comment:
Note that this was now working because that part is only executed in mag 3D mode